### PR TITLE
Add TraceID(ctx context.Context)

### PR DESCRIPTION
**Pull request by arugal**
- Related issues
#25

### CODE REVIEW

**Comment by wu-sheng** [comment]
One suggestion, the documentation should update for this.

File: trace.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ import (
 
 const errParameter = tool.Error("parameter are nil")
 
+var emptyTraceID = []int64{0, 0, 0}
```
Comment (hanahmily) [inline trace.go]: How about declaring it as constant

Comment (hanahmily) [reply trace.go]: And make it accessible from another package

Comment (arugal) [reply trace.go]: > How about declaring it as constant

Constants cannot be array.

Ref: 
1. https://stackoverflow.com/questions/13137463/declare-a-constant-array
1. https://blog.golang.org/constants

Comment (arugal) [reply trace.go]: I would like to modify it to keep it consistent with the Java agent.
1. activeSpan == nil -> N/A
1. activeSpan == NoopSpan -> [Ignored Trace]
1. activeSpan == segmentSpan -> part1.part2.part3

Comment (wu-sheng) [reply trace.go]: Does go2sky could work in no backend service? I don't remember this part, could you recheck?

Comment (arugal) [reply trace.go]: > Does go2sky could work in no backend service? 

`Reporter` should work when not registered,`TestTrace_TraceID ` covers all three

Comment (wu-sheng) [reply trace.go]: OK. If that is supported, it makes sense. But then, trace Id should be a string.

**Comment by arugal** [comment]
> One suggestion, the documentation should update for this.

provided

File: trace.go (hunk 1 above)
Comment (arugal) [reply trace.go]: Done.

**Pull request merged**